data/account/participationRegistry.go

```diff
@@ -46,13 +46,18 @@ type ParticipationRecord struct {
 	LastVote               basics.Round
 	LastBlockProposal      basics.Round
 	LastCompactCertificate basics.Round
-	RegisteredFirst        basics.Round
-	RegisteredLast         basics.Round
+	EffectiveFirst         basics.Round
+	EffectiveLast          basics.Round
 
 	// VRFSecrets
 	// OneTimeSignatureSecrets
 }
 
+// IsZero returns true if the object contains zero values.
+func (r ParticipationRecord) IsZero() bool {
+	return r == (ParticipationRecord{})
```

Comment:
I'm wondering if creating a global empty object and comparing it to that one would get executed faster.

Reply:
I suppose it couldn't hurt. If it does help maybe we should do that for Address as well